go-controller/pkg/config/utils.go

```diff
@@ -271,3 +273,68 @@ func ContainsJoinIP(ip net.IP) bool {
 	}
 	return false
 }
+
+// masqueradeIP represebts the masqueradeIPs used by the masquerade subnets for host to service traffic
+type MasqueradeIP struct {
+	V4OVNMasqueradeIP               net.IP
+	V6OVNMasqueradeIP               net.IP
+	V4HostMasqueradeIP              net.IP
+	V6HostMasqueradeIP              net.IP
+	V4HostETPLocalMasqueradeIP      net.IP
+	V6HostETPLocalMasqueradeIP      net.IP
+	V4DummyNextHopMasqueradeIP      net.IP
+	V6DummyNextHopMasqueradeIP      net.IP
+	V4OVNServiceHairpinMasqueradeIP net.IP
+	V6OVNServiceHairpinMasqueradeIP net.IP
+}
+
+// allocateV4/6MasqueradeIPs allocates the masqueradeIPs based off of the passed in masqueradeSubnet (.0)
+// it does this by cascading down from the initial ip down to the .5 currently (more masqueradeIps may be added in the future)
+
+func allocateV4MasqueradeIPS(masqueradeSubnet net.IP, masqueradeIPs MasqueradeIP) error {
+	masqueradeIPs.V4OVNMasqueradeIP = iputils.NextIP(masqueradeSubnet)
+	if masqueradeIPs.V4OVNMasqueradeIP == nil {
+		return fmt.Errorf("error setting V4OVNMasqueradeIP")
```

Comment:
In the error messages you return, I would include the starting address you got as input `masqueradeSubnet`. Same thing for the v6 function below.